recipes/nanopub/meta.yaml

```
@@ -0,0 +1,48 @@
+{% set name = "nanopub" %}
+{% set version = "1.2.10" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 63748da4e28e85cab3927c3a608adb943a7cec9a234c0fcf974902892f75178f
+
+build:
+  number: 0
+  skip: True  # [win]
```

This looks like pure Python so you should be able to make it `noarch:  python`. You'll also need to list the entry point for that:
```suggestion
  noarch: python
  entry_points:
    - setup_nanopub_profile=nanopub.setup_nanopub_profile:main
```

---

Thanks for reviewing @dopplershift! 🙏  Unfortunately we have a dependency on [`nanopub-java`](https://github.com/Nanopublication/nanopub-java), the installation for windows is a bit squeaky and breaks on the conda-forge builds. Unfortunately we have no resources to make the windows install smoother any time soon, although there is an  open issue for it: https://github.com/fair-workflows/nanopub/issues/118. 

Preferably we would like to have this on conda without windows support, hence the non-noarch conda package.

---

Feel free to merge if you agree.

---

So this installs `nanopub-java` outside of conda by itself?

---

Yes @dopplershift you are right! See [setup instructions](https://nanopub.readthedocs.io/en/latest/getting-started/setup.html#nanopub-java-dependency) and [setup.py](https://github.com/fair-workflows/nanopub/blob/3f7565f70068c2aa22d5f374f51b72f0e060bda5/setup.py#L35)

---

Is there any reason not to make a conda-forge recipe for `nanopub-java` and just depend on it like normal?

---

If that is easy to do that sounds like the proper solution. 

Unfortunately I don't maintain `nanopub-java`, and I don't speak java. To be honest we just want to get `nanopub-python` on conda-forge with as little effort as possible, the main reason that many of the potential users (scientists) only use conda to get their python packages. So, the reason is: little time to invest in this project and laziness.

So, if necessary we can create a `conda-forge` recipe for `nanopub-java`, preferably we stick with the solution we currently have.

---

@dopplershift any update?

---

I haven't checked everything yet but if the only reason you don't want Windows is a missing dependency you can also keep this package noarch and use the `__unix` virtual package: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html

Also, conda-forge _really_ prefers having all dependencies listed under requirements and installed with the package. But I can understand that it might be difficult here.

---

Thanks for your reaction @BastianZim ! I will try and update the PR using the `__unix` virtual package.